internal/services/datafactory/data_factory_integration_runtime_azure_ssis_resource.go

```diff
@@ -869,6 +883,15 @@ func flattenDataFactoryIntegrationRuntimeAzureSsisCatalogInfo(ssisProperties *da
 		dualStandbyPairName = *ssisProperties.DualStandbyPairName
 	}
 
+	var pricingTier, elasticPoolName string
+	if strings.HasPrefix(string(ssisProperties.CatalogPricingTier), "ELASTIC_POOL") {
+		if result := regexp.MustCompile(`(?:name=\")([^&%\\\/?]{0,127}[^\s.&%\\\/?])(?:\")`).FindStringSubmatch(string(ssisProperties.CatalogPricingTier)); len(result) > 1 {
+			elasticPoolName = result[1]
+		}
```

Comment:
Can we put this piece of code to a separate function (like `parseElasticPool()`)? Also we shall simplify it to something like below:

```
{
    matches := regexp.MustCompile(`^ELASTIC_POOL\(name="(.+)"\)$`).FindStringSubmatch(s)
    if len(matches) != 2 {
      return "", false
   }
   return matches[1], true
}
```